src/pybind/mgr/dashboard/services/cephfs.py

```diff
@@ -161,6 +161,32 @@ def rm_dir(self, path):
         logger.info("Removing directory: %s", path)
         self.cfs.rmdir(path)
 
+    def recursive_rm_dir(self, path):
+        """
+        Recursive remove a directory
+        :param path: The path of the directory.
+        """
+        if path == os.sep:
+            raise Exception('Cannot remove root directory "/"')
+        if not self.dir_exists(path):
+            return
+        logger.info("Recursive removing directory: %s", path)
+        handle = self.cfs.opendir(path)
+        while True:
+            de = self.cfs.readdir(handle)
+            if de is None:
+                break
+            dent_name = de.d_name.decode()
+            if dent_name == "." or  dent_name == "..":
+                continue
+            new_path = path + "/" + dent_name
+            if de.is_dir():
+                self.recursive_rm_dir(new_path)
+            else:
+                self.cfs.unlink(new_path)
```

Comment:
I'd perform this in a 2-stage process:
1. Go through the file tree collecting the files/dirs to be deleted
2. Then perform the deletion from that collection (and since you have the total number, you can update the progress in the `Task.set_progress(deleted_items/total_items)`)